diff -up libunistring-1.0/config.h.in.omv~ libunistring-1.0/config.h.in
--- libunistring-1.0/config.h.in.omv~	2022-01-05 17:48:27.634438724 +0100
+++ libunistring-1.0/config.h.in	2022-01-05 17:48:36.317449888 +0100
@@ -1741,7 +1741,7 @@
    _GL_ATTRIBUTE_DEALLOC_FREE declares that the function returns pointers that
    can be freed via 'free'; it can be used only after declaring 'free'.  */
 /* Applies to: functions.  Cannot be used on inline functions.  */
-#if _GL_GNUC_PREREQ (11, 0)
+#if _GL_GNUC_PREREQ (11, 0) && !defined (__clang__)
 # define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 #else
 # define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up libunistring-1.0/gnulib-m4/gnulib-common.m4.omv~ libunistring-1.0/gnulib-m4/gnulib-common.m4
--- libunistring-1.0/gnulib-m4/gnulib-common.m4.omv~	2022-01-05 17:47:46.078385275 +0100
+++ libunistring-1.0/gnulib-m4/gnulib-common.m4	2022-01-05 17:48:03.742407998 +0100
@@ -176,7 +176,7 @@ AC_DEFUN([gl_COMMON_BODY], [
    _GL_ATTRIBUTE_DEALLOC_FREE declares that the function returns pointers that
    can be freed via 'free'; it can be used only after declaring 'free'.  */
 /* Applies to: functions.  Cannot be used on inline functions.  */
-#if _GL_GNUC_PREREQ (11, 0)
+#if _GL_GNUC_PREREQ (11, 0) && !defined (__clang__)
 # define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 #else
 # define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up libunistring-1.0/lib/stdlib.in.h.omv~ libunistring-1.0/lib/stdlib.in.h
--- libunistring-1.0/lib/stdlib.in.h.omv~	2022-01-05 17:49:32.369521925 +0100
+++ libunistring-1.0/lib/stdlib.in.h	2022-01-05 17:49:38.972530407 +0100
@@ -103,7 +103,7 @@ struct random_data
    that can be freed by passing them as the Ith argument to the
    function F.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if __GNUC__ >= 11
+# if __GNUC__ >= 11 && !defined (__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up libunistring-1.0/lib/string.in.h.omv~ libunistring-1.0/lib/string.in.h
--- libunistring-1.0/lib/string.in.h.omv~	2022-01-05 17:49:56.589553036 +0100
+++ libunistring-1.0/lib/string.in.h	2022-01-05 17:50:10.682571135 +0100
@@ -71,7 +71,7 @@
    that can be freed by passing them as the Ith argument to the
    function F.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if __GNUC__ >= 11
+# if __GNUC__ >= 11 && !defined (__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up libunistring-1.0/lib/unistring/cdefs.h.omv~ libunistring-1.0/lib/unistring/cdefs.h
--- libunistring-1.0/lib/unistring/cdefs.h.omv~	2022-01-05 17:48:48.145465094 +0100
+++ libunistring-1.0/lib/unistring/cdefs.h	2022-01-05 17:48:58.851478855 +0100
@@ -88,7 +88,7 @@
 # define _UC_HAS_C_ATTRIBUTE(attr) 0
 #endif
 
-#if _UC_GNUC_PREREQ (11, 0)
+#if _UC_GNUC_PREREQ (11, 0) && !defined (__clang__)
 # define _UC_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 #else
 # define _UC_ATTRIBUTE_DEALLOC(f, i)
diff -up libunistring-1.0/lib/wchar.in.h.omv~ libunistring-1.0/lib/wchar.in.h
--- libunistring-1.0/lib/wchar.in.h.omv~	2022-01-05 17:49:11.451495048 +0100
+++ libunistring-1.0/lib/wchar.in.h	2022-01-05 17:49:20.674506899 +0100
@@ -88,7 +88,7 @@
    that can be freed by passing them as the Ith argument to the
    function F.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if __GNUC__ >= 11
+# if __GNUC__ >= 11 && !defined (__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up libunistring-1.0/tests/stdio.in.h.omv~ libunistring-1.0/tests/stdio.in.h
--- libunistring-1.0/tests/stdio.in.h.omv~	2022-01-05 17:48:10.384416541 +0100
+++ libunistring-1.0/tests/stdio.in.h	2022-01-05 17:48:18.891427482 +0100
@@ -95,7 +95,7 @@
    that can be freed by passing them as the Ith argument to the
    function F.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if __GNUC__ >= 11
+# if __GNUC__ >= 11 && !defined (__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
